### DISCUSSION TITLE
Does GitHub Actions (cloud) allow running of ARM based instances?

### DISCUSSION THREAD

**Discussion by joshuarobs** [community/community#25319] [Actions]
<p>When we run code online for testing with CI/CD, is there an option to make it so that our code can run on Ubuntu on ARM? I’m assuming that all existing OS options are all on x86</p>

**Comment by airtower-luna** [comment]
<p>I’m not aware of any directly supported options. If you’re willing to build a somewhat complex workflow you could either set up a VM, or (for better performance) cross-compile and then use Qemu user space emulation to test ARM binaries in the runner system. Here’s a <a href="https://wiki.debian.org/QemuUserEmulation" rel="nofollow noopener">tutorial for Debian</a>, it should be possible to adapt that to Ubuntu.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/joshuarobs">@joshuarobs</a>,</p>
<p>Glad to see you in Github Community Forum!<br>
Yes, the Ubuntu hosted runners provided by Github are all ‘x86_64’ architecture.  If you’d like to run code on Ubuntu on ARM, you can create a self-hosted runner, and invoke it in the workflow.<br>
<a href="https://help.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners</a>.</p>
<p>Thanks.</p>

**Comment by joshuarobs** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Thanks for clarification. Also can you run your own self hosted runners with a free account of GitHub or is this only a premium feature?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/joshuarobs">@joshuarobs</a>,</p>
<p>Free account can use self-hosted runner.<br>
And it’s recommended to use self-hosted runner with private repository to avoid malicious code from fork. Please refer to the <a href="https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories" rel="nofollow noopener">link</a> for more details.</p>
<p>Thanks</p>

**Comment by alexiswl** [comment]
Would be great to ARM being directly supported by GitHub actions, I've found qemu to be quite slow for 32-bit ARM versions. Don't get me wrong, the fact that qemu can still build a 32-bit ARM image on 64-bit amd64 architecture is incredibly impressive! 

ARM support from GitHub actions would mean being able to [swarm](https://docs.docker.com/engine/swarm/) your build compilations, but also generally test how code performs on ARM instances.

**Reply by laoshanxi** [reply]
Yes, I already use Github native ARM runner now:
https://github.com/laoshanxi/app-mesh/blob/main/.github/workflows/build-push-appmesh-image.yml

**Reply by bkiseu** [reply]
@laoshanxi Is it possible to do the same (using github actions to build native arm on ubuntu) but instead push the image to ec2 instead of pushing to docker?

**Reply by laoshanxi** [reply]
> @laoshanxi Is it possible to do the same (using github actions to build native arm on ubuntu) but instead push the image to ec2 instead of pushing to docker?

It's quite flexible, you can push to any registry, you can refer to above workflow and this workflow:

https://github.com/laoshanxi/app-mesh/blob/main/.github/workflows/publish-github-release.yml

**Reply by bkiseu** [reply]
Thanks. The above workflow really helped. I was having a problem with the docker/build-push-action@v4 action as well but https://github.com/docker/build-push-action/issues/576 helped me.

**Comment by rblaine95** [comment]
Looks like Github will be adding ARM based runners in the near future 🥳 🚀 🎊 
- https://github.com/github/roadmap/issues/836
- https://resources.github.com/devops/accelerate-your-cicd-with-arm-and-gpu-runners-in-github-actions/

**Reply by Sn0rt** [reply]
aha, I wonder if GitHub has plans to provide some free arm runner quotas to open source projects?

**Reply by rblaine95** [reply]
@Sn0rt, I'm not aware of any official statement regarding this, but I see no reason why not

**Reply by crohr** [reply]
If you can't wait, you can check out [RunsOn](https://runs-on.com)!

**Comment by ghost** [comment]
Still no public access and no information about free plans and public repositories - [Arm-based hosted runners are coming to GitHub Actions! 🚢 #73040](https://github.com/orgs/community/discussions/73040).

**Reply by nebuk89** [reply]
We are working on public repos and free plans, watch this space <3

**Reply by avarayr** [reply]
It's been almost 6 months guys, this sucks :(

**Reply by ghost** [reply]
Most probably we are on the finish line, hopefully - [ARM hosted runners for OSS/free plans [Public Preview] #970](https://github.com/github/roadmap/issues/970)

<img width="311" alt="399931145-6111148d-437f-47e2-b76b-ef78d80a3dab" src="https://github.com/user-attachments/assets/f884fe27-258c-457a-8599-239b078594eb" />

**Reply by nebuk89** [reply]
https://github.blog/changelog/2026-01-29-arm64-standard-runners-are-now-available-in-private-repositories/

**Reply by nebuk89** [reply]
Sorry it took a long time 😬 we are hurrying up on a bunch now <3 thank you for the patience!
